sync/syncer.go

```diff
@@ -104,14 +105,12 @@ type Syncer struct {
 	txpool               TxMemPool
 	atxpool              AtxMemPool
 	lValidator           LayerValidator
-	currentLayer         types.LayerID
-	currentLayerMutex    sync.RWMutex
 	validatingLayer      types.LayerID
 	validatingLayerMutex sync.Mutex
 	SyncLock             uint32
 	startLock            uint32
 	forceSync            chan bool
-	LayerCh              timesync.LayerTimer
+	syncTimer            *time.Ticker
```

Comment:
Make sure the disadvantages of time.Ticker (aggregate delay, resources release) are no bother for your use case/you handle it.
In addition, you can use an interface here and then you will be able to test it better with a "ticker" mock.

Reply:
cool idea, maybe in a later revision when the need arrives, for now this aupports all our current test cases